sys/arch/init_mach.c

```diff
@@ -208,7 +208,7 @@ _getmch (void)
 	fputype = detect_fpu();
 #ifndef WITH_68080
 	/* own SFP-004 test */
-	sfptype = detect_sfp();
+	sfptype = (mcpu < 0x20) ? detect_sfp() : 0;
```

Comment:
See also this thread: https://mikro.naprvyraz.sk/mint/freemint-list/Week-of-Mon-20170306/000798.html ... I made the same mistake back in 2015 and fixed it only as late as 2017. :)

Reply:
Hmmn, yeah I am unhappy with this change but decided to put it forward to spark a conversation in the hopes someone has a better idea.

Essentially what is happening:

68LC060, on a non-Atari hardware:
Correctly detects no FPU (now, with the other fix in this PL)
SFP is detected, thus the "fpu" boolean is set to true some lines after.

Mints context switcher for 060 does not check what kind of FPU there is.
If the "fpu" boolean is true then it will assume a 060 FPU and proceed with fsave/frestore = bad time.

The proper way may be to change the context switcher code to check the kind of FPU in case it's an SFP but that feels like building code for a scenario that is very hard to test on hardware.
It could check for machine_unkown, but I suppose you'd still run into crashing the context switcher if you manage to put a 68LC060 in a MegaST.

I don't really know what the best solution is here.

Reply:
> Unless we rather change the context switcher to deal with LC + SFP.

The fpu variable (which is checked by the context switcher) is set at https://github.com/freemint/freemint/blob/cc58689a29c73e0a6e5cf4b02fc0d5422c664562/sys/arch/init_mach.c#L217-L218

Note that this explicitly checks for `> 1`, which does not include an SFP. So i would assume the context switcher should work in this case (saving/restoring the registers of an SFP is imho not possible)

Reply:
> saving/restoring the registers of an SFP is imho not possible)

On a second though, that is actually problematic, when there are two programs running that both use the SFP. But no idea how to fix that, theoretically you should prevent context switches until the SFP is done, and the app has fetched the results. But that can hardly be done in the kernel.

Reply:
I came to the same conclusion back in 2015/2017. Basically running two SFP-aware applications is not supported.